internal/codespaces/api/api.go

```diff
@@ -514,8 +515,8 @@ var errProvisioningInProgress = errors.New("provisioning in progress")
 // It may return success or an error, or errProvisioningInProgress indicating that the operation
 // did not complete before the GitHub API's time limit for RPCs (10s), in which case the caller
 // must poll the server to learn the outcome.
-func (a *API) startCreate(ctx context.Context, repoID int, machine, branch, location string) (*Codespace, error) {
-	requestBody, err := json.Marshal(startCreateRequest{repoID, branch, location, machine})
+func (a *API) startCreate(ctx context.Context, repoID, idleTimeoutMinutes int, machine, branch, location string) (*Codespace, error) {
+	requestBody, err := json.Marshal(startCreateRequest{repoID, idleTimeoutMinutes, branch, location, machine})
```

Comment:
Lets switch to initializing the `startCreateRequest` struct with named fields instead of relying on argument order to map to fields.